src/db.rs

```diff
@@ -85,8 +83,73 @@ pub fn insert_connection(connection: Connection) -> Result<Uuid> {
     Ok(nwuuid)
 }
 
+pub fn insert_policy(policy: &policy::Policy) -> Result<u64> {
+    let mut dbconn = get_dbconn()?;
+
+    let allowed_digests_json = serde_json::to_string(&policy.allowed_digests)?;
+    let allowed_policy_json = serde_json::to_string(&policy.allowed_policies)?;
+    let allowed_build_ids_json = serde_json::to_string(&policy.allowed_build_ids)?;
+
+    let mysqlstr = format!(
+        "INSERT INTO policy (allowed_digests, allowed_policies, min_fw_api_major, min_fw_api_minor,
+                               allowed_build_ids, create_date, valid)
+                               VALUES({:?}, {:?}, {}, {}, {:?}, NOW(), 1)",
+        allowed_digests_json,
+        allowed_policy_json,
+        policy.min_fw_api_major,
+        policy.min_fw_api_minor,
+        allowed_build_ids_json
+    );
```

Comment:
I'm not sure I trust `{:?}` to do string-escaping for SQL. On top of my head I can't think of an attack, but you'd want either some escaping provided by the mysql crate, or (= better) use query with `?` and pass arguments separately (like you do in `insert_secret` below).

Reply:
I agree and did attempt to code this function initially the same way.  The issue that I ran into is that after the transaction is committed, the last insert id gets set to 0, which looks like a default response.  If I attempt to grab the last insert id prior to committing, I run into ownership problems for the connection.  I am sure that there is  a way to get around this, but I wasn't able to see a solution immediately.  I will keep looking, but have the sense that sqlx may be a better way to screen for injection attacks.

Reply:
Since `Conn` implements `Queryable`, I think that you should be able to do:

```rust
dbconn.exec_drop("INSERT INTO policy (allowed_digests, allowed_policies, min_fw_api_major, min_fw_api_minor,
                               allowed_build_ids, create_date, valid) 
                               VALUES (?, ?, ?, ?, ?, NOW(), 1)", 
                 (allowed_digests_json, allowed_policy_json, policy.min_fw_api_major, 
                  policy.min_fw_api_minor, allowed_build_ids_json));
Ok(dbconn.last_insert_id())
```

but I haven't tested it. If my suggestion doesn't work, just keep as-is.